### ISSUE TITLE
Extend across stylesheets —

### ISSUE THREAD

**Issue by tusharmath** [cssinjs/jss#283]
```js
const jss = require('jss').default
const preset = require('jss-preset-default').default
jss.setup(preset())

// Style 1
jss.createStyleSheet({
  'flexRow': { display: 'flex'}
})

// Style 2
jss.createStyleSheet({
  button: {
    fontSize: 12,
    extend: 'flexRow'
  }
})

console.log(jss.sheets.toString())
```

Current
```css
flexRow {
  display: flex;
}
.button-2242923773 {
  font-size: 12px;
}
```

Expected

```css
flexRow {
  display: flex;
}
.button-2242923773 {
  font-size: 12px;
  display: flex;
}
```

How can I solve this problem? Is there a way around it?

**Comment by kof** [comment]
You can do that by extending plain objects.

```
const a = {color: 'red'}
const styles = {
  button: {
    extend: a
  }
}
```

**Comment by trusktr** [comment]
@kof Hi, I'm currently doing

```js
import inputStyle from './input.style'

export default Object.assign({}, inputStyle, {
    //
})
```

but I think it might be better to start using `extend` and/or `compose`?

Do you use `Object.assign` vs `extend`? What are your recommendations?

**Comment by trusktr** [comment]
Alright, I switched to using

```js
import inputStyle from './input.style'

export default {
    foo: {
        extend: inputStyle.foo
    }
}
```

which I think is more flexible, and less verbose. Loving it! This is my first time working with (in this case) React components and extending styles from base styles. I love it so far.

I made a `StyleableComponent` class that extends from `React.Component`, and does basically the same thing as what your `injectStylesheet` does, except we access `this.classes` on the instance instead of `this.props.classes` being passed in.
